tsdb/engine/tsm1/cache_test.go

```diff
@@ -223,7 +225,7 @@ func TestCacheLoader_LoadSingle(t *testing.T) {
 		Values: values,
 	}
 
-	if err := w.Write(entry); err != nil {
+	if err := w.Write(mustMarshalEntry(entry)); err != nil {
```

Comment:
Yes, I see that now:

"update WALSegmentWriter.Write to take the compressed bytes so that encoding and compression can occur before the call to write (while we don't hold the WAL lock)"